### ISSUE TITLE
Forward OpenTracing Headers in Plugin HTTP Calls

### ISSUE THREAD

**Issue by JHK** [traefik/traefik#10560]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you expect to see?

I'm currently developing a plugin for Traefik (v3.0.0-rc3) where `tracing.otlp` is enabled. The plugin requires calling an external authentication service, during which it needs to append authentication headers including OpenTracing headers for proper traceability. However, the current implementation lacks a straightforward method to forward or attach OpenTracing headers (such as trace ID and a potentially self-generated span ID) to these outbound requests, ensuring the correct parent-child span relationship.

Within the plugin, I use a `net.http.Client` for HTTP calls. I am looking for a way to either:
1. Have access to necessary tracing information within the `context.Context.Value` passed to the plugin, or
2. Embed the logic of attaching tracing headers directly into the HTTP client, ensuring all outbound requests carry the correct tracing headers automatically.

As of now, I haven't located any tracing information within the `context.Context` or documentation that could guide me on how to achieve this.

**Comment by JHK** [comment]
I did not extract the trace context from the request context. I just did a blunt logging `println("ctx: %+v", a.Ctx)` to see into what is available and haven't found the tracing context in there.

**Comment by rtribotte** [comment]
@JHK Could you try with OTel library as in the ForwardAuth middleware:https://github.com/traefik/traefik/blob/fc897f6756a3a6ae9e3acd604a71ae4f4a675245/pkg/middlewares/auth/forward.go#L137-L148

**Comment by JHK** [comment]
I tried (see https://github.com/JHK/auth-demo/pull/1 / [usage](https://github.com/JHK/auth-demo?tab=readme-ov-file#usage)), but cannot load the `github.com/traefik/traefik/v3/pkg/tracing` dependency:

```
failed to craete Yaegi intepreter: failed to import plugin code \"github.com/JHK/auth-demo\": 1:21: import \"github.com/JHK/auth-demo\" error: plugins-local/src/github.com/JHK/auth-demo/demo.go:10:2: import \"github.com/traefik/traefik/v3/pkg/tracing\" error: unable to find source related to: \"github.com/traefik/traefik/v3/pkg/tracing\"
```

**Comment by rtribotte** [comment]
Hello @JHK,

Sorry for the late answer, and the maybe misleading hint.
I was implying to do what is done in Traefik, but it is better to copy the logic and not depend on Traefik packages.
Yaegi comes with some limitations, and the less it has to interpret the better, having said that, maybe OTel itself is not "usable" with Yaegi (I did not give it a try).

**Comment by JHK** [comment]
I will give it a try to copy some logic from `github.com/traefik/traefik/v3/pkg/tracing` into the plugin and see if that would do the trick and come back to you. Otherwise the initial phrasing of the ticket would stay valid I'd say.

**Comment by JHK** [comment]
> maybe OTel itself is not "usable" with Yaegi

Yeah, it doesn't seem like its working. We tried to inline the traefik packages, but got an error that `go.opentelemetry.io/otel` is missing:

```
"failed to craete Yaegi intepreter: failed to import plugin code \"github.com/JHK/auth-demo\": 1:21: import \"github.com/JHK/auth-demo\" error: plugins-local/src/github.com/JHK/auth-demo/demo.go:10:2: import \"github.com/JHK/auth-demo/tracing\" error: plugins-local/src/github.com/JHK/auth-demo/tracing/tracing.go:7:2: import \"github.com/JHK/auth-demo/tracing/opentelemetry\" error: plugins-local/src/github.com/JHK/auth-demo/tracing/opentelemetry/opentelemetry.go:15:2: import \"go.opentelemetry.io/otel\" error: unable to find source related to: \"go.opentelemetry.io/otel\""
```

Also further inlining of code didn't work as haven't figured out a way to redefine that constant `currentSpanKey` to run a `ctx.Value(currentSpanKey)` ([source](https://github.com/open-telemetry/opentelemetry-go/blob/main/trace/context.go)).
